### DISCUSSION TITLE
Adding I18N causes permission problems in the middleware

### DISCUSSION THREAD

**Discussion by XiaoNode** [nuxt-modules/i18n#1459] [Q&A]
hey there:
some permission settings exist in my project，When accessing the userinfo page, you must log in first, otherwise you will jump to the auth(login) page. This part is done using middleware;

When I use official examples(readme.md) in projects，permission settings Something went wrong，When accessing the userinfo page，console tips:

`[vue-router] Route with name 'auth' does not exist`

What's going on？

Project address： https://github.com/XiaoNode/Nuxt2-temp

**Comment by rchl** [comment]
You have to use the APIs that this module provides for resolving routes. For example in https://github.com/XiaoNode/Nuxt2-temp/blob/b213f0e6d72089959f27fd97880f4e409d3469f8/middleware/auth.js#L13 you have to do:
```js
    redirect(app.localePath("auth"));
```
where `app` is from the context passed to the plugin function.

**Reply by XiaoNode** [reply]
get it, thks!
